packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -356,6 +359,12 @@ class DatePickerDialog extends StatefulWidget {
   ///    Flutter.
   final String? restorationId;
 
+
+  /// Called when the [DatePickerDialog] is toggled between [DatePickerEntryMode.calendar],[DatePickerEntryMode.input]
```

Comment:
For API doc (not code), we try and keep the line length to roughly 80 characters. 

It would be good to append a second sentence or two (after the usual blank line) to explain what this callback is for; what a typical use case is.

Reply:
Wrapped the API Doc in next two lines